updater/lib/dependabot/updater/operations.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require "dependabot/updater/operations/update_all_versions"
+
+# This module is responsible for determining which Operation a Job is requesting
+# the Updater to perform.
+#
+# The design goal for this module is to make these classes easy to understand,
+# maintain and extend so we can eventually support community-contributed
+# alternatives or ecosystem-specific implementations.
+#
+# Consider the following guidelines when working on Operation classes:
+#
+# - Operations act on the principal of least knowledge prefering to cherry pick
+#   arguments from the Dependabot::Job so their contract is explicit.
+#
+# - Operations *should not have optional parameters*, prefer to create a new
+#   class instead of adding branching to an existing one.
+#
+# - Operations should prefer to share logic by composition, there is no base
+#   class. We want to avoid implicit or indirect behaviour as much as possible.
```

Comment:
😄 I'm appreciating these design principles. A comment seems right. I wonder if, _in the future_, we could enforce this with linting so that someone would have to be very intentional about going against these principles.